ngraph/frontend/ir/src/ir_deserializer.cpp

```diff
@@ -0,0 +1,745 @@
+    std::shared_ptr<ngraph::Function> ngraph_function;
+    if (!name.compare("body")) {
+        auto body_node = m_node.child(name.c_str());
+        if (body_node.empty()) {
+            IE_THROW() << "TensorIterator has no body.";
+        }
+        ngraph_function = parse_function(m_node.child(name.c_str()), m_weights);
+    } else if (!name.compare("net")) {
+        ngraph_function = parse_function(m_node, m_weights);
+    } else {
+        IE_THROW() << "Error: not recognized adapter name: " << name << ".";
+    }
+    adapter.set(ngraph_function);
+}
+
+std::shared_ptr<ngraph::Function> XmlDeserializer::parse_function(const pugi::xml_node& root,
+                                                                  const ov::Weights& weights) {
+    // OV_ITT_SCOPE_CHAIN(FIRST_INFERENCE, taskChain, itt::domains::V10Reader_RT, "V10Parser", "Parse");
+
+    struct FunctionNodes {
+        ngraph::ParameterVector parameters;
+        ngraph::ResultVector results;
+        ngraph::NodeVector all;
+        ngraph::SinkVector sinks;
+    };
+
+    struct edge {
+        size_t fromLayerId, fromPortId, toPortId;
+    };
+    struct node_params {
+        pugi::xml_node xml;
+        GenericLayerParams params;
+    };
+
+    std::map<size_t /*layer-id*/, node_params> params;
+
+    std::vector<size_t /*layer-id*/> outputs;
+    std::unordered_set<std::string> opName;
+
+    // Read all layers and store their parameters in params map
+    FOREACH_CHILD (node, root.child("layers"), "layer") {
+        auto node_param = parseGenericParams(node);
+        if (opName.find(node_param.name) != opName.end() && node_param.type != "Result")
+            IE_THROW() << "Invalid IR! " << node_param.name << " name is not unique!";
+        opName.insert(node_param.name);
+        params[node_param.layerId] = {node, node_param};
+        if (node_param.type == "Result" || node_param.type == "Assign") {
+            outputs.push_back(node_param.layerId);
+        }
+    }
+
+    std::map<size_t /*to-layer-id*/, std::vector<edge>> edges;
+    std::map<size_t, std::shared_ptr<ngraph::Node>> id_to_node;
+
+    // Read all edges and store them for further usage
+    FOREACH_CHILD (_ec, root.child("edges"), "edge") {
+        size_t fromLayer = XMLParseUtils::GetUIntAttr(_ec, "from-layer");
+        size_t fromPort = XMLParseUtils::GetUIntAttr(_ec, "from-port");
+        size_t toLayer = XMLParseUtils::GetUIntAttr(_ec, "to-layer");
+        size_t toPort = XMLParseUtils::GetUIntAttr(_ec, "to-port");
+        edges[toLayer].push_back({fromLayer, fromPort, toPort});
+    }
+
+    // Run DFS starting from outputs to get nodes topological order
+    std::set<size_t> used;
+    std::vector<size_t> order;
+    std::function<void(size_t)> dfs = [&edges, &order, &used, &dfs](const size_t id) {
+        if (used.count(id))
+            return;
+        used.insert(id);
+        for (auto& edge : edges[id]) {
```

Comment:
`const auto&`?

Reply:
This code was marked as changed because I've split the giant file with all code to the smaller files. So all this deserialization engine was copy-pasted as is. The only change here is rt_info support.